runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -2351,7 +2489,295 @@ J9::Z::TreeEvaluator::checkcastEvaluator(TR::Node * node, TR::CodeGenerator * cg
       }
    else
       {
```

Comment:
Not related to this PR but we should work on this `TODO` to remove if/else in the evaluator. 

We utilize the common API to generate list of the tests we want to perform. Conditions such as  `comp->getOption(TR_OptimizeForSpace) || comp->getOption(TR_DisableInlineCheckCast)` should be tested in `calculateInstanceOfOrCheckCastSequences` [https://github.com/eclipse/openj9/blob/e72b823f79156a2fa6229d3a635e8cee2c92eb39/runtime/compiler/codegen/J9TreeEvaluator.cpp#L686] where it returns only `NULLTEST` and `HelperCall` .

Reply:
Agreed. Will open an issue for this as well.

Reply:
Opened up #6190 to track this.